File: 0014-longest-common-prefix/0014-longest-common-prefix.php
class Solution {

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs) {
        if (count($strs) === 0) {
            return "";
        }
        $preffix = $strs[0];
        for ($i = 1; $i < count($strs); $i++) {
            while (!str_contains($strs[$i], $preffix)) {
                $preffix = substr($preffix, 0, strlen($preffix) - 1);
                if ($preffix === "") {
                    return $preffix;
                }
            }
        }
        return $preffix;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    res := ""
    for i, _ := range strs[0] {
        for _, word := range strs {
            if i == len(word) || word[i] != strs[0][i] {
                return res
            }
        }
        res += string(strs[0][i])
    }
    return res
}